app/assets/javascripts/views/question_form.js

```diff
@@ -91,6 +91,9 @@
 
     // populate and show the modal
     $("#create-option-set .modal-body.option-set").load(self.params.new_option_set_path, function(){
+      var multilevel_checkbox = $("#option_set_multi_level");
+      self._can_be_multi_level() ? multilevel_checkbox.removeAttr('disabled') : multilevel_checkbox.attr('disabled', 'disabled')
```

Comment:
I see what you're doing here but I think that enabling/disabling the checkbox should be the responsibility of the OptionSetForm JS class. The question form really shouldn't know anything about that box, or whether an option set can be multilevel. Principle of least responsibility.

Instead I would pass a param to the OptionSetForm class indicating the question type for which the option set is getting created. Then let it decide how to handle itself.

You can do this by passing a GET param along with the .load call, which you can then pass to the OptionSetForm object on instantiation. (in option_sets/_form.html.erb).

Make sense?